src/services/data_model_engine.js

```diff
@@ -173,12 +173,20 @@ export default class DataModelEngine {
     if (bundle === null) {
       throw new NotFoundError(`No bundle with id = ${bundleId} found`);
     }
-    return bundle;
+    let metadata = await this.entityRepository.getBundleMetadata(bundleId);
+    if (metadata === null) {
+      return bundle;
+    }
+    metadata = pick(metadata, 'bundleId');
+    return {...bundle, metadata};
   }
 
   async getBundleMetadata(bundleId) {
-    const {metadata} = await this.getBundle(bundleId);
-    return {...metadata, bundleId};
+    const metadata = await this.entityRepository.getBundleMetadata(bundleId);
+    if (metadata === null) {
+      throw new NotFoundError(`No bundle metadata for bundleId = ${bundleId} found`);
```

Comment:
```suggestion
      throw new NotFoundError(`No metadata found for bundleId = ${bundleId}`);
``` may be better (too much bundles for 5 words :))